src/software/ai/navigator/path_manager/path_objective.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+#include <vector>
+
+#include "software/ai/navigator/obstacle/obstacle.h"
+
+/**
+ * Path objective is used to pass information from navigator
+ * to path manager to plan paths
+ */
+class PathObjective
+{
+   public:
+    PathObjective(const Point start, const Point end, const double current_velocity,
+                  const std::vector<Obstacle> &obstacles, RobotId robot_id)
+        : start(start),
+          end(end),
+          current_velocity(current_velocity),
+          obstacles(obstacles),
+          robot_id(robot_id)
+    {
+    }
+
+    PathObjective(const PathObjective &other)
+        : start(other.start),
+          end(other.end),
+          current_velocity(other.current_velocity),
+          obstacles(other.obstacles),
+          robot_id(other.robot_id)
+    {
+    }
+
+    const RobotId robot_id;
```

Comment:
I get that these values can't be changed once set due to being `const`, so they are technically safe to leave as public to be accessed. 

This is the first time I'm seeing a datastructure that can do this, so just want to take this chance to decide on if we're ok with this in general as a convention? Every other class we have uses functions to expose values, so I'm debating the tradeoff between less work (don't need to write accessor functions) vs consistency with our other classes.

Reply:
It's like accessing a struct and all the values are const so it's just holds data.

Reply:
I would prefer being consistent with our other classes, as Jon once said: "mental cycles > line count"

Reply:
I did...?

Reply:
Sorry, i misquoted: "Clarity > line count" (https://github.com/UBC-Thunderbots/Software/pull/993#discussion_r339391739)

Reply:
Ah yes. I agree with the wisdom of my past self

Reply:
In this case, I don't think the extra lines improve clarity

Reply:
But they do slightly affect how data is accessed, since they're public members rather than functions which is different from the rest of our code. I don't feel _too_ strongly about this but would prefer consistency.

Reply:
This is a bigger issue than this instance so I'm making a new issue to review how we do this across the whole repo: #1011